sse.go

```diff
@@ -255,7 +265,6 @@ func (s *Server) dispatch() {
 			close(s.addClient)
 			close(s.removeClient)
 			close(s.closeChannel)
-			close(s.shutdown)
```

Comment:
Why did this move to shutdown function? The existing behaviour made sense to me.
1. Shutdown function is called which sends an event to the shudown channel
2. dispatch function receives the shutdown event and then closes the channel as no further events are expected

Reply:
As you can see, in this PR all the serving goroutines (ServerHTTP) use reading from the s.shutdown channel. At the same time,  the`dispatch` function also reads from this channel. It means, that if we just send some value to this channel, we don't know where this value will be read: in the `dispatch` function or in the one of running `ServeHTTP` functions. If we close this channel, then all the functions will receive default values by reading from this channel.

Reply:
@insano10 Please, take a look at another PR (#2) where I added a unit test for the entire shutdown flow. This test fails with the current implementation and succeeds with the updated implementation.

Reply:
Thank you for the explanation, that makes sense now :)